File: ssort.js
class SelectionSort {
    static sort(a) {
        for (let i = 0; i < a.length - 1; i++) {
            let min = i;
            for (let j = i + 1; j < a.length; j++) {
                if (a[min] > a[j]) {
                    min = j;
                }
            }

            let temp = a[min];
            a[min] = a[i];
            a[i] = temp;
        }

        console.log("The sorted array is:");
        for (let i = 0; i < a.length; i++) {
            console.log(a[i] + "\t");
        }
    }
}

let a = [1, 4, 5, 76, 89, 33, 55, 0, 21, 56];
SelectionSort.sort(a);

File: ssort.cpp
#include <iostream>
using namespace std;

class SelectionSort {
public:
    static void sort(int a[], int size) {
        for (int i = 0; i < size - 1; i++) {
            int min = i;
            for (int j = i + 1; j < size; j++) {
                if (a[min] > a[j]) {
                    min = j;
                }
            }

            int temp = a[min];
            a[min] = a[i];
            a[i] = temp;
        }

        cout << "The sorted array is:\n";
        for (int i = 0; i < size; i++) {
            cout << a[i] << "\t";
        }
    }
};

int main() {
    int a[] = {1, 4, 5, 76, 89, 33, 55, 0, 21, 56};
    int size = sizeof(a) / sizeof(a[0]);

    SelectionSort::sort(a, size);

    return 0;
}
